core/domain/caching_services.py

```diff
@@ -29,47 +29,52 @@
 
 memory_cache_services = models.Registry.import_cache_services()
 
-# NOTE: Namespaces and sub-namespaces cannot contain ':'.
+# NOTE: Namespaces and sub-namespaces cannot contain ':' because this is used as
+# an internal delimiter for cache keys.
+
 # This namespace supports sub-namespaces which are identified by the stringified
 # version number of the explorations within the sub-namespace. The value for
 # each key in this namespace should be a serialized representation of an
-# Exploration. If we are not storing multiple versions of an Exploration in the
-# cache, the sub-namespace is not required to differentiate the keys of multiple
-# versions of an exploration and can therefore be None.
+# Exploration. There is also a special sub-namespace represented by the empty
+# string; this sub-namespace handles the explorations where only one version is
```

Comment:
Are you sure about this? I thought it handles the current/latest version of the exploration. How did you come to the above conclusion?

(Similarly below.)